libcaf_core/caf/policy/work_stealing.hpp

```diff
@@ -85,7 +85,8 @@ class work_stealing : public unprofiled {
       return nullptr;
     }
     // roll the dice to pick a victim other than ourselves
-    size_t victim = d(self).rengine() % (p->num_workers() - 1);
+    std::uniform_int_distribution<size_t> uniform(0, p->num_workers() - 2);
```

Comment:
`N` workers excluding self are `N - 1` workers. We should select from these `N - 1` workers, not `N` workers, otherwise we may have a chance to select self, and thus a collision. Among `[0..(N - 2)]` are the `N - 1` indices we need. The self index  may be within `[0..(N - 2)]` or not. In the latter case, things are fine trivially. In the former case, suppose self index is `X`, things are fine if the generated index is not `X`. If it is `X`, it should actually be `N - 1`. This is because we are not supposed to generate `X`, and `N - 1` would never be generated for it is out of the range `[0..(N - 2)]`.